TFNetworkLayer.py

```diff
@@ -1933,7 +1933,9 @@ def get_out_data_from_opts(
       out_type["shape"] = list(out_type["shape"])
       if out_type["shape"][axis_wo_batch] is not None:
         out_type["shape"][axis_wo_batch] = len(range(out_type["shape"][axis_wo_batch])[dim_slice])
-      if axis_wo_batch == len(out_type["shape"]) - 1 and not out_type["sparse"]:
+      if "feature_dim_axis" not in out_type:
+        out_type["feature_dim_axis"] = len(out_type["shape"])
+      if axis_wo_batch == out_type["feature_dim_axis"] - 1 and not out_type["sparse"]:
```

Comment:
The calculation is not always correct (only if batch axis is before feature dim axis).